src/main/groovy/io/openliberty/tools/gradle/tasks/AbstractServerTask.groovy

```diff
@@ -731,8 +748,91 @@ abstract class AbstractServerTask extends AbstractTask {
+            //Merge server.env props with generated serverEnvPath
+            mergedProperties = combineServerEnvProperties(serverEnvProps, configuredProps);
+            // writeServerEnvProperties(envFile, mergedProperties);
+        }
+
+        if (server.serverEnvFile != null && server.serverEnvFile.exists()) {
+            if (serverEnvPath != null) {
+                logger.warn("The " + serverEnvPath + " file is merged with the " + server.serverEnvFile.getCanonicalPath() + " file.")
+            }
+            Properties configuredServerEnvProps = convertServerEnvToProperties(server.serverEnvFile);
+            //Merge with either default server.env or with what has already been merged if
+            Properties propsToMergeWith = mergedProperties.isEmpty() ? serverEnvProps : mergedProperties;
+
+            mergedProperties = (Properties) combineServerEnvProperties(propsToMergeWith, configuredServerEnvProps);
+        }
+
+        if(!mergedProperties.isEmpty()) {
+            writeServerEnvProperties(envFile, mergedProperties);
+        }
+    }
+
+    private void setServerEnvHelper(File envFile, String serverEnvPath, Properties configuredProps) {
+        if ((server.env != null && !server.env.isEmpty()) || !envProjectProps.isEmpty()) {
+            if (serverEnvPath != null) {
+                logger.warn("The " + serverEnvPath + " file is overwritten by inlined configuration.")
+            }
+            writeServerEnvProperties(envFile, configuredProps)
+            serverEnvPath = "inlined configuration"
+        } else if (server.serverEnvFile != null && server.serverEnvFile.exists()) {
+            if (serverEnvPath != null) {
+                logger.warn("The " + serverEnvPath + " file is overwritten by the " + server.serverEnvFile.getCanonicalPath() + " file.")
+            }
+            Files.copy(server.serverEnvFile.toPath(), envFile.toPath(), StandardCopyOption.REPLACE_EXISTING)
+            serverEnvPath = server.serverEnvFile.getCanonicalPath()
+        }
+    }
+
+    private Properties convertServerEnvToProperties(File serverEnv) {
+        Properties serverEnvProps = new Properties();
+
+        if ((serverEnv == null) || !serverEnv.exists()) {
+            return serverEnvProps;
+        }
+
+        BufferedReader bf = new BufferedReader(new FileReader(serverEnv));
+        String line;
+        while((line = bf.readLine()) != null) {
+            
+            //Skip comments
+            if(!line.startsWith("#")) {
+                String[] keyValue = line.split("=", 2);
+                String key = keyValue[0];
+                String value = keyValue[1];
+
+                serverEnvProps.put(key,value);
+            }
+        }
+
+        return serverEnvProps;
+    }
+
+    private Properties combineServerEnvProperties(Properties properties, Properties projectProperties) {
         Properties combinedEnvProperties = null
+        //Create new properties object
```

Comment:
Please change line 834 to create a new Properties object. Then change line 838 to do a `combinedEnvProperties.putAll(projectProperties)` and line 846 to do a `combinedEnvProperties.putAll(properties)`. Then you can remove line 840.

Reply:
Had missed this from last review. Should be in the next commit 👍